Sources/MapboxNavigation/VisualInstruction.swift

```diff
@@ -135,50 +159,62 @@ extension VisualInstruction {
-
-        if let image = lightLanesImage, let darkImage = darkLanesImage, image.size.width > lanesImageMaxSize.width {
-            let aspectRatio = lanesImageMaxSize.width / image.size.width
+        
+        let lightTraitCollection = UITraitCollection(traitsFrom: [
+            UITraitCollection(userInterfaceIdiom: .carPlay),
+            UITraitCollection(userInterfaceStyle: .light)
+        ])
+        var lightLanesImage = lanesViewImage(drivingSide: side,
+                                             direction: direction,
+                                             useableColor: LaneView.appearance(for: lightTraitCollection).primaryColor,
+                                             unuseableColor: LaneView.appearance(for: lightTraitCollection).secondaryColor,
+                                             size: CGSize(width: CGFloat(laneComponents.count) * lanesImageMaxSize.height, height: lanesImageMaxSize.height),
+                                             scale: scale)
+        
+        let darkTraitCollection = UITraitCollection(traitsFrom: [
+            UITraitCollection(userInterfaceIdiom: .carPlay),
+            UITraitCollection(userInterfaceStyle: .dark)
+        ])
+        var darkLanesImage = lanesViewImage(drivingSide: side,
+                                            direction: direction,
+                                            useableColor: LaneView.appearance(for: darkTraitCollection).primaryColor,
+                                            unuseableColor: LaneView.appearance(for: darkTraitCollection).secondaryColor,
+                                            size: CGSize(width: CGFloat(laneComponents.count) * lanesImageMaxSize.height, height: lanesImageMaxSize.height),
+                                            scale: scale)
+        
+        if let lightImage = lightLanesImage,
+           let darkImage = darkLanesImage,
+           lightImage.size.width > lanesImageMaxSize.width {
+            let aspectRatio = lanesImageMaxSize.width / lightImage.size.width
             let scaledSize = CGSize(width: lanesImageMaxSize.width, height: lanesImageMaxSize.height * aspectRatio)
-            lightLanesImage = image.scaled(to: scaledSize)
+            lightLanesImage = lightImage.scaled(to: scaledSize)
             darkLanesImage = darkImage.scaled(to: scaledSize)
         }
-        if let image = lightLanesImage, let darkImage = darkLanesImage {
-            return CPImageSet(lightContentImage: image, darkContentImage: darkImage)
+        
+        if let lightImage = lightLanesImage,
+           let darkImage = darkLanesImage {
+            // `CPImageSet` applies `lightContentImage` for dark appearance and `d`arkContentImage`
+            // for light appearance, because of this `darkImage` is set for `lightContentImage` parameter.
+            return CPImageSet(lightContentImage: darkImage, darkContentImage: lightImage)
```

Comment:
This looks like a system bug, basically colors are inverted. `CPImageSet` was also deprecated in iOS 13. I'll create a separate ticket to find a suitable replacement.